# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1486410819 -3600
#      Mon Feb 06 20:53:39 2017 +0100
# Branch stable
# Node ID 50e3a3e962c032900bd9eb45867ce9311592f49c
# Parent  8419d24ab8f705d81f0ff5c4a12961a039da17ca
make editor save its widget settings again when octave quits

* main-window.cc (construct): add the missing connection between the
  aboutToQuit signal and the editors save_settings slot

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -1369,16 +1369,18 @@ main_window::construct (void)
                history_window, SLOT (save_settings ()));
       connect (qApp, SIGNAL (aboutToQuit ()),
                file_browser_window, SLOT (save_settings ()));
       connect (qApp, SIGNAL (aboutToQuit ()),
                doc_browser_window, SLOT (save_settings ()));
       connect (qApp, SIGNAL (aboutToQuit ()),
                workspace_window, SLOT (save_settings ()));
       connect (qApp, SIGNAL (aboutToQuit ()),
+               editor_window, SLOT (save_settings ()));
+      connect (qApp, SIGNAL (aboutToQuit ()),
                this, SLOT (prepare_to_exit ()));
       connect (qApp, SIGNAL (aboutToQuit ()),
                shortcut_manager::instance, SLOT (cleanup_instance ()));
       // QSettings are saved upon deletion (i.e., cleanup_instance)
       connect (qApp, SIGNAL (aboutToQuit ()),
                resource_manager::instance, SLOT (cleanup_instance ()));
 
       connect (qApp, SIGNAL (focusChanged (QWidget*, QWidget*)),
